rosetta-source/src/main/java/cdm/event/common/processor/DocumentationHelper.java

```diff
@@ -25,25 +29,36 @@ public class DocumentationHelper {
     private final RosettaPath rosettaPath;
     private final List<Mapping> mappings;
     private final SynonymToEnumMap synonymToEnumMap;
+    private final ExecutorService executor;
 
     public DocumentationHelper(RosettaPath rosettaPath, MappingContext context) {
         this.rosettaPath = rosettaPath;
         this.mappings = context.getMappings();
+        this.executor = context.getExecutor();
         this.synonymToEnumMap = context.getSynonymToEnumMap();
     }
 
-    public List<LegalAgreement> getDocumentation(Path synonymPath) {
+    public List<LegalAgreement> getDocumentation(Path synonymPath, MappingContext mappingContext) {
         List<LegalAgreement> documentation = new ArrayList<>();
-        getMasterAgreement(synonymPath).ifPresent(documentation::add);
-        getMasterConfirmation(synonymPath).ifPresent(documentation::add);
-        getBrokerConfirmation(synonymPath).ifPresent(documentation::add);
-        getCreditSupportAgreement(synonymPath).ifPresent(documentation::add);
-        getConfirmation(synonymPath).ifPresent(documentation::add);
-        getOtherAgreement(synonymPath).ifPresent(documentation::add);
+        getMasterAgreement(synonymPath, mappingContext).ifPresent(documentation::add);
+        getMasterConfirmation(synonymPath, mappingContext).ifPresent(documentation::add);
+        getBrokerConfirmation(synonymPath, mappingContext).ifPresent(documentation::add);
+        getCreditSupportAgreement(synonymPath, mappingContext).ifPresent(documentation::add);
+        getConfirmation(synonymPath, mappingContext).ifPresent(documentation::add);
+        getOtherAgreement(synonymPath, mappingContext).ifPresent(documentation::add);
         return documentation;
     }
 
-    private Optional<LegalAgreement> getMasterAgreement(Path synonymPath) {
+    private Optional<LegalAgreement> getMasterAgreement(Path synonymPath, MappingContext mappingContext) {
+
+        // Check if the synonymPath ends with "masterAgreement" using mappings
+        boolean isMasterAgreement = mappingContext.getMappings().stream()
```

Comment:
You don't need to pass in `MappingContext`.  You can replace `mappingContext.getMappings().stream()` with `mappings.stream()`

Reply:
Hi Hugo thanks for the feedback, will change that and recontribute it

Reply:
I have recontributed this part, mappingContext.getMappings().stream() with mappings.stream(). But we will need as a parameter the MappingContext mappingContext in the methods called on getDocumentation() since we will be calling PartyMappingHelper.getInstance(mappingContext) so we can get the counterparties and then add it to the contractualParty of the legalAgreement.